content/blog/tikv-3.0ga-release-notes.md

```diff
@@ -0,0 +1,55 @@
+
+* **Expanding our coprocessor.** With lots of new functionality, improved APIs, vector operations, and better ways to work with data, our coprocessor continues to evolve, allowing for better, more efficient ways to work with your data.
+
+* **Enhancing operator friendliness.** Human or machine, we've empowered our operators to get more out of TiKV by unifying our log format, adding new features to `tikv-ctl`, adding even more in depth metrics, and serving HTTP based metrics. This makes TiKV easier to operate, inspect, and enjoy.
+
+* **Refining request types.** While TiKV previously had commands like `BatchGet`, 3.0 brings a new `BatchCommands` request type. We've made taught TiKV how to handle batches of requests of differing kinds, leading to less data on the wire and better performance. We also added support for raw reversed scanning, and `Insert` semantics on prewrite.
+
+* **Reducing write amplication.** We were inspired by the great ideas from [WiscKey](https://www.usenix.org/system/files/conference/fast16/fast16-papers-lu.pdf) and implemented a feature we dubbed 'Titan'. It works best with larger (>1 KB) values.
+
+You can see all the changes in detail [here](https://github.com/tikv/tikv/blob/release-3.0/CHANGELOG.md).
+
+## Improved Performance
+
+Using [`go-ycsb`](https://github.com/pingcap/go-ycsb) we benchmarked TiKV 3.0.0-rc.2 against TiKV 2.1.12. We benchmarked a cluster of 3 TiKV nodes and 1 PD node. They were running on public cloud machines with 16 cores, 32 GB of RAM, a 400 GB NVMe SSD, and a 10Gbps network. We used the default settings of TiKV and PD for this.
+
+In order to achieve high concurrency and properly stress the system, we ran YCSB across three machines and set the following:
+
+* 1KB Value Size
+* 100 Field Length
+* 10 Field Count
+* 3000 Thread Count (for each machine, so a total of 9000)
+
+| Workload              | operation | 2.1.12 Avg Latency | 3.0.0-rc.2 Avg Latency | 2.1.12 QPS | 3.0.0-rc.2 QPS | QPS Diff |
+|-----------------------|-----------|--------------------|------------------------|------------|----------------|----------|
+| A 50% read/50% update | read      | 4.908              | 1.971                  | 40645.7    | 43159.2        | +6.18%   |
+|                       | update    | 206.649            | 197.140                | 40635.7    | 43186.5        | +6.28%   |
+| B 95% read/5% update  | read      | 28.121             | 13.967                 | 247164.1   | 542027.0       | +119.30% |
+|                       | update    | 132.953            | 43.307                 | 12988.6    | 28632.4        | +120.44% |
+| C 100% read           | read      | 31.056             | 14.583                 | 282752.8   | 606659.5       | +114.55% |
+| Load (10 MB)          | insert    | 198.764            | 217.901                | 44743.5    | 40819.9        | -8.77%   |
```

Comment:
Why is load slower 🤣

Reply:
It was AWS machines so might have been some jitter or noisy neighbor

Reply:
Ah, it's not 10MB. It's 10,000,000 pieces of 1KB data.

Reply:
please reproduce it and find the reason.

Reply:
The performance for workload a is not as high as i expected in 3.0, can we re-test it?

Reply:
Definitely! :)